interfaces/repo.go

```diff
@@ -265,6 +265,71 @@ func (r *Repository) RemoveSlot(snapName, slotName string) error {
 	return nil
 }
 
+// ResolveConnect resolves potentially missing plug or slot names and returns a
+// fully populated connection reference.
+func (r *Repository) ResolveConnect(plugSnapName, plugName, slotSnapName, slotName string) (*ConnRef, error) {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	if plugSnapName == "" {
+		return nil, fmt.Errorf("cannot resolve connection, plug snap name is empty")
+	}
+	if plugName == "" {
+		return nil, fmt.Errorf("cannot resolve connection, plug name is empty")
+	}
+	// Ensure that such plug exists
+	plug := r.plugs[plugSnapName][plugName]
+	if plug == nil {
+		return nil, fmt.Errorf("cannot resolve connection, plug %q from snap %q, no such plug", plugName, plugSnapName)
+	}
+
+	if slotSnapName == "" {
+		// Use the core snap if the slot-side snap name is empty
+		switch {
+		case r.slots["core"] != nil:
+			slotSnapName = "core"
+		case r.slots["ubuntu-core"] != nil:
+			slotSnapName = "ubuntu-core"
```

Comment:
Didn't we want to avoid hardcoding os snap name alltogeter? John's latest changes should introduce OS snap name getter to the state, so we could get it from there and pass to ResolveConect?